src/libsyntax/diagnostics/plugin.rs

```diff
@@ -125,8 +125,18 @@ pub fn expand_register_diagnostic<'cx>(ecx: &'cx mut ExtCtxt,
             span,
             span,
             sym,
-            Vec::new(),
-            Vec::new()
+            vec![
+                // #[allow(non_snake_case)]
```

Comment:
While I generally have nothing against this, I'm wondering if we should do an `in_macro` in the lint instead.

Reply:
I think the diagnostic is already not reported in macros? Is this just something funky with syntax extensions?

Reply:
`with_empty_ctxt` already creates an identifier with a dummy span, so won't the lint get triggered anyways? Seems eaiser to me to just allow the lint 😅

Reply:
So... if it's a dummy, why is the lint triggered at all? Did you maybe do this change before you added the dummy check and the `allow` is now not necessary anymore?

Reply:
Thanks! That also explains why the lint suddenly started getting reported.